File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        lastzIndex=-1
        left=0
        right=0
        count=0
        for i in range(len(nums)):
            if(nums[i]==0):
                left = lastzIndex+1
                lastzIndex = i
            right = i
            count = max(count, right-
            left+1)
        return count -1
        

File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        
        int f = 1;
        int r=0,l=0;
        int count = 0, Max_count=0;
        while(r<nums.size()){
            if(nums[r]==1){
                count = r-l+1;
                r++;
            }
            else if(nums[r]==0 && f==1){
                count = r-l+1;
                f = 0;
                r++;
            }
            else if(nums[r]==0 && f==0){
                if(nums[l]==0) f=1;
                l++;
            }
            Max_count = max(count, Max_count);
        }
        return Max_count - 1;
    }
};